### DISCUSSION TITLE
Advised "roles" insecure?

### DISCUSSION THREAD

**Discussion by levino** [pocketbase/pocketbase#5486] [Q&A] [answered]
I am trying to add a "role" to my users. However the [recommended](https://pocketbase.io/docs/collections/#auth-collection) way seems to be insecure without additional api rules for the collection. If I simply add a field "role" to the auth collection, the users can just change the value, can they not?

**Comment by ganigeorgiev** [comment, marked as answer]
That depends on your auth collection Create/Update API rules and your app requirements.

If you don't want to allow users to be able to assign or change their role you could either overwrite the submitted value with a hook or use for example the [`:isset` filter modifier](https://pocketbase.io/docs/api-rules-and-filters/#isset-modifier), e.g.:
```js
@request.data.role:isset = false
```

I'll consider explaining it better with the new upcoming v0.23 docs.

**Reply by levino** [reply]
I figured it out from other posts too and was just coming here to add my findings. Thanks for the quick reply. I think it is a very good idea to add this info to that section. Lots of people might not see this problem. Without the rule, any user was really able to promote their roles.

**Reply by ganigeorgiev** [reply]
> I figured it out from other posts too and was just coming here to add my findings.

Then write as it is because by saying "a bit funky" without actually elaborating or suggesting how to fix it is not fine and I'm getting tired reading such complaints. Why ask questions when you know the answer? This is a hobby project and I work on it on volunteer basis. If you don't like it, just use something else and move on.

**Reply by levino** [reply]
I am sorry. I think you misunderstood. I meant that I came back to add the answer which I had found out while further investigating and researching and then found that you already had given it. I did not mean to be rude. When I created the original question, I really was lost. Probably my comment was unnecessary. Sorry.

**Reply by levino** [reply]
I updated the OP to be a bit more neutral.
